### PR TITLE
Updating Functions!

### PR DESCRIPTION
**Pull request by devmike123**

### CODE REVIEW

File: src/functions/usdchanneltype.md [hunk 1]
```diff
@@ -6,18 +6,29 @@ description: Returns channel type
...
+| channel ID | The id of the channel | number | no |
+
 
 #### Types:
 
 * `text` =&gt; normal text channel
 * `voice` =&gt; voice channel
 * `category` =&gt; category
 * `news` =&gt; announcement channel
+* `stage` =&gt; stage channel
```
Comment (USERSATOSHI) [inline src/functions/usdchanneltype.md:27]: missing more channel types
```js
$djseval[Object.keys(d.util.channelTypes).map(x => x.toLowerCase());yes]
```

eval this to see all available channel types

File: src/functions/usdcheckcondition.md [hunk 2]
```diff
@@ -6,21 +6,19 @@ description: Checks if the given condition is true or false
...
-#### Options
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| condition | Whether the provided condition is met | alphanumeric | yes |
 
-* Value 1 - The value we're comparing to value 2
-* Value 2 - The value we're comparing to value 1
 
-#### Conditions
+#### Conditional Operators
```
Comment (USERSATOSHI) [inline src/functions/usdcheckcondition.md:22]: also add && , ||

File: src/functions/usdcheckcontains.md [hunk 3]
```diff
@@ -1,31 +1,25 @@
...
+### Fields
 
-* Text - The text we're checking
-* Word - The word we're checking to see if it's in the &lt;text&gt;
-* Word 2 -The second word we're checking to see if it's in the &lt;text&gt;
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| text | The text to be checked | string | yes |
+| word1 | The word whose presence is to be checked in the text | string | yes |
+| word2 | The word whose presence is to be checked in the text | string | no |
```
Comment (USERSATOSHI) [inline src/functions/usdcheckcontains.md]: can be changed to  [...words] array of words to match

File: src/functions/usdclear.md [hunk 4]
```diff
@@ -8,27 +8,23 @@ description: >-
...
+```php
+$clear[amt;userFilter?;return_amount?;channelID?]
+```
+### Fields
 
-## Options
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| amt | The amount of messages to be deleted | number | yes |
+| userFilter | The user the bot is deleting messages from |User ID/everyone|yes|
```
Comment (USERSATOSHI) [inline src/functions/usdclear.md]: unPins , bot, everyone , userId

these are the available filters

File: src/functions/usdclient.md [hunk 5]
```diff
@@ -6,7 +6,16 @@ description: A compact funcion with 17 different functionalities.
...
+$client[option]
+```
+
+### Fields
+
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| option | The option to be used | string | yes |
 
 #### Usable Options:
```
Comment (USERSATOSHI) [inline src/functions/usdclient.md]: eval $client[json]
and this should give you all available options

File: src/functions/usdcolorrole.md [hunk 6]
```diff
@@ -6,21 +6,19 @@ description: Changes the color of the specified role's ID
```
Comment (USERSATOSHI) [inline src/functions/usdcolorrole.md]: this function has been renamed to $setRoleColor

File: src/functions/usdcommandinfo.md [hunk 7]
```diff
@@ -2,29 +2,27 @@
...
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| command name | The name of command | string | yes |
+| property | The property to be used|string|yes|
 
 #### Properties
 
 * Name - The name of the command
 * Code - The code of the command
 * Aliases - The aliases of the command
```
Comment (USERSATOSHI) [inline src/functions/usdcommandinfo.md]: there is whitelist
executeAt

and custom properties that user defines in cmd

File: src/functions/usdconcattextsplit.md [hunk 8]
```diff
@@ -2,19 +2,19 @@
...
 
-#### Options
+### Fields
 
-* Text - The text we're concatenating
-* Separator - The separator separating each value
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| text | The text to be separated | string | yes |
+| separator | The separator used to separate the values |alphanumeric|yes|
```
Comment (USERSATOSHI) [inline src/functions/usdconcattextsplit.md]: usage is $concatTextSplit[...text]

example :
```js
//[ "ok", "no" , "hi", "bye" ]
$concattextSplit[hi;bye]
$textSplit[ok,no;,]
```

File: src/functions/usdcooldown.md [hunk 9]
```diff
@@ -4,24 +4,22 @@ description: Set a cooldown for a command
...
+```
 
-Raw Usage: `$cooldown[time;error]`
+### Fields
 
-#### Options
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| time | The time of cooldown | alphanumeric | yes |
+| error | The error to be displayed when cooldown is active |string|yes|
```
Comment (USERSATOSHI) [inline src/functions/usdcooldown.md:20]: add available %options% that are present in cooldown functions

[link](https://github.com/AkaruiDevelopment/aoi.js/blob/af90c0947d41949edeb43759159aa3d1f3578e07/src/functions/Funcs/util/cooldown.js#L30)

File: src/functions/usdcreatevar.md [hunk 10]
```diff
@@ -2,23 +2,20 @@
...
+$createVar[varName:varValue;varName:varValue?...]
+```
+### Fields
 
-* Var Name - The name of the variable
-* Var Value - The default value of the variable
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| Var1 | The name of var with value | name:value | yes |
+| Var2 | The name of var with value |name:value|no|
```
Comment (USERSATOSHI) [inline src/functions/usdcreatevar.md]: usage is `$createVar[table;var:value;var:value;...]`

File: src/functions/usdcreatewebhook.md [hunk 11]
```diff
@@ -6,27 +6,23 @@ description: Creates a webhook to the given channel ID
...
-Raw Usage:  `$createWebhook[channel ID;name;avatar url;returnWebhook ID/Token (yes/no);separator (optional)]`
+### Fields
 
-#### Options
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| channel ID | The id of the channel  | number | yes |
+| name | The name of webhook |string|yes|
+|avatar_url|The url of the avatar of webhook|url|yes|
+|return ID|Whether the webhook is or token is to be returned|yes/no|yes|
```
Comment (USERSATOSHI) [inline src/functions/usdcreatewebhook.md:21]: there is no returnId

File: src/functions/usdday.md [hunk 12]
```diff
@@ -2,19 +2,17 @@
...
+#### Usage 
+```php
+$day[returnDayOfTheWeek?]
+```  
+### Fields
 
-This function has 1 optional field
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| returnDayOfTheWeek | Whether to return the day of the week  | yes/no | no|
```
Comment (USERSATOSHI) [inline src/functions/usdday.md]: there are no fields in $day

File: src/functions/usdhasanyrole.md [hunk 13]
```diff
@@ -4,20 +4,32 @@ description: checks if the user has at least one of the required roles
...
+| user ID | The id of the user whose roles are to be checked | number | yes |
+| role ID | The id of the role which is to be checked | number | yes |
+
+## Example
 
 ```javascript
 bot.command({
     name: "hasrole",
     code: `
     $hasAnyRole[773353338854572073;773353339496169513]
```
Comment (USERSATOSHI) [inline src/functions/usdhasanyrole.md]: example is wrong

File: src/functions/usdhasrole.md [hunk 14]
```diff
@@ -4,17 +4,29 @@ description: 'Check if the User, has the Roles or Not.'
 
 # $hasRole
```
Comment (USERSATOSHI) [inline src/functions/usdhasrole.md]: this function doesn't exists

File: src/functions/usdhoistedrole.md [hunk 15]
```diff
@@ -25,12 +31,12 @@ code: `$hoistedRole`
...
 ```
 
-With optional fields
+- With optional fields
 
 ```javascript
 bot.command({
 name: 'hoistedRole',
-code: `$hoistedRole[$mnetioned[1]]`
+code: `$hoistedRole[$mnetioned[1];$guildID;mention]`
```
Comment (USERSATOSHI) [inline src/functions/usdhoistedrole.md]: spelling  of mentioned is wrong

File: src/functions/usdjsonrequest.md [hunk 16]
```diff
@@ -4,12 +4,25 @@ description: Retrieve a JSON format (api)
...
+### Fields
+
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| api | The api from which the specified property is to be retrieved | link | yes |
+| property | The property that is to be retrieved from the api | alphanumeric | no |
+| error | The error to be displayed if the property wasn't found | alphanumeric | no |
+
+## Example
+
```
Comment (USERSATOSHI) [inline src/functions/usdjsonrequest.md]: missing headers field

accepts normal or json

normal :

```
A:hi;B:Bye
```

json:

```json
{ "A" : "hi", "B" : "bye" }
```

File: src/functions/usdkick.md [hunk 17]
```diff
@@ -4,16 +4,25 @@ description: Kick using user ID
...
+| user ID | The id of the user to be kicked | number | yes |
+| guild ID | The id of the guild | number | no |
+| reason | The reason why the user was kicked | alphanumeric | yes |
 
+## Example
 ```javascript
 bot.command({
 name: "kick", 
-code: `$kick[$mentioned[1]]`
+code: `$kick[$mentioned[1];Spam]`
```
Comment (USERSATOSHI) [inline src/functions/usdkick.md]: example is wrong ( missing guildId part )

File: src/functions/usdclearsongqueue.md [hunk 18]
```diff
@@ -6,9 +6,14 @@ description: Clears all songs in queue besides 1st song in queue.
```
Comment (USERSATOSHI) [inline src/functions/usdclearsongqueue.md:6]: this function is renamed to $clearQueue

File: src/functions/usdloopqueue.md [hunk 19]
```diff
@@ -4,7 +4,14 @@ description: Loop the current songs in the queue. Returns whether or not it's lo
 
 # $loopQueue
```
Comment (USERSATOSHI) [inline src/functions/usdloopqueue.md]: this function is merged into $loopMode[queue]

File: src/functions/usdloopsong.md [hunk 20]
```diff
@@ -1,8 +1,16 @@
+---
+description: Loops the current song in the queue.
+---
 # $loopSong
```
Comment (USERSATOSHI) [inline src/functions/usdloopsong.md]: this function is merged into $loopMode[song]

**Review by USERSATOSHI — CHANGES_REQUESTED** [review]

File: src/functions/usdclient.md (hunk 5 above)
Comment (devmike123) [reply src/functions/usdclient.md]: Not working

Comment (USERSATOSHI) [reply src/functions/usdclient.md]: image
uh.....

**Comment by devmike123** [comment]
@USERSATOSHI ready for merge!

File: src/functions/usdclient.md [hunk 21]
```diff
@@ -36,6 +36,7 @@ $client[option]
 * `token` =&gt; returns the bot's token. More information: see [$clientToken](usdclienttoken.md)
```
Comment (USERSATOSHI) [inline src/functions/usdclient.md]: options needs an update

File: src/functions/usdcommandinfo.md [hunk 22]
```diff
@@ -16,9 +16,15 @@ $commandInfo[command_name;property]
...
-* Aliases - The aliases of the command
+* name - The name of the command
+* code - The code of the command
+* aliases - The aliases of the command
+* executeAt - Returns both if command can be used in `both` dm and guild, `guild` if in guild and `dm` if in dm.
+* whitelist - Returns boolean whether or not the command is whitelisted for blacklisted users.
+* nonPrefixed - Returns boolean whether or not if the author message starts with command name.
+* usage - Returns the usage of the command if set.
+* type - Returns the type of the command. [Here's the list.](../options/commandTypes.md)
+* info - Returns the info of the command if any.
```
Comment (USERSATOSHI) [inline src/functions/usdcommandinfo.md]: something like

[custom property]: value added in cmd
is better cause usage is not by default a valid property in aoijs cmd

File: src/functions/usdloopmode.md [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+### Fields
+
+| Field | Description | Type | Required |
+| :--- | :--- | :--- | :--- |
+| mode | The mode based on which the song/queue will be looped/unlooped | string | yes |
+
+#### Modes
+- `song` - To loop/unloop current song in the queue.
+- `queue` - To loop/unloop all songs in the queue.
+
```
Comment (USERSATOSHI) [inline src/functions/usdloopmode.md:23]: there is a none to disable too

**Review by USERSATOSHI — CHANGES_REQUESTED** [review]

**Comment by devmike123** [comment]
@USERSATOSHI ready for merge

**Review by USERSATOSHI — APPROVED** [review]
LGTM

**Pull request merged**